src/styles/main.scss

```diff
@@ -0,0 +1,70 @@
+@import '../styles/utils/variables';
+
+body {
+  margin: 0;
+
+  display: flex;
+  align-items: center;
+  text-align: center;
+  flex-direction: column;
+  justify-content: center;
+  height: 100vh;
+
+  font-family: Arial, sans-serif;
+  font-size: $font-size;
+}
+
+.calendar {
+  margin-left: 14px;
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap;
+  padding: $padding;
+
+  width: calc(#{$day-size} * #{$week} + #{$padding} * 2);
+
+  &__day {
+    box-sizing: border-box;
+    justify-content: center;
+    align-items: center;
+
+    width: $day-size;
+    height: $day-size;
+
+    background-color: $default-color;
+    border: $border;
+    transition-duration: 0.5s;
+
+    &:hover {
+      cursor: pointer;
+      background-color: $hover-color;
+      transform: translateY(-20px);
+    }
+  }
+
+  @for $monthLength from 28 through 31 {
+    &--month-length-#{$monthLength} {
+      .calendar__day {
+        display: none;
+      }
+
+      @for $day from 1 through 31 {
+        .calendar__day:nth-child(#{$day})::before {
+          content: '#{$day}';
+        }
+      }
+
+      .calendar__day:nth-child(-n + #{$monthLength}) {
+        display: flex;
+      }
+    }
```

Comment:
The loop from 28 through 31 is used to generate modifiers for different month lengths, but it seems like the loop that sets the `content` property for the `.calendar__day::before` is running from 1 through 31 inside each month length modifier. This will result in unnecessary CSS rules, as days beyond the month length will never be displayed. Instead, the loop for setting `content` should be limited to the current month length `$monthLength`.